src/Command/Push/PushCodeCommand.php

```diff
@@ -34,10 +40,157 @@ protected function configure() {
+    $this->push($output_callback, $artifact_dir, $commit_hash);
+    $this->checklist->completePreviousItem();
 
     return 0;
   }
 
+  /**
+   * Prepares a directory to build the artifact.
+   *
+   * @param \Closure $output_callback
+   * @param string $artifact_dir
+   * @param string $vcs_url
+   * @param string $vcs_path
+   *
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function prepareDir(Closure $output_callback, string $artifact_dir, string $vcs_url, string $vcs_path): void {
+    $fs = $this->localMachineHelper->getFilesystem();
+
+    $this->logger->info("Removing $artifact_dir if it exists");
+    $fs->remove($artifact_dir);
+
+    $this->logger->info("Initializing Git in $artifact_dir");
+    $process = $this->localMachineHelper->executeFromCmd("git clone --depth 1 --branch $vcs_path $vcs_url $artifact_dir", $output_callback, NULL, $this->output->isVerbose());
+    if (!$process->isSuccessful()) {
+      throw new AcquiaCliException('Failed to clone repository from the Cloud Platform: {message}', ['message' => $process->getErrorOutput()]);
+    }
+
+    $this->logger->info('Global .gitignore file is temporarily disabled during artifact builds.');
+    $this->localMachineHelper->executeFromCmd('git config --local core.excludesFile false', $output_callback, $artifact_dir, $this->output->isVerbose());
+    $this->localMachineHelper->executeFromCmd('git config --local core.fileMode true', $output_callback, $artifact_dir, $this->output->isVerbose());
+  }
+
+  protected function build(Closure $output_callback, string $artifact_dir): void {
+    // @todo generate a deploy identifier
+    // @see https://git.drupalcode.org/project/drupal/-/blob/9.1.x/sites/default/default.settings.php#L295
+
+    // @todo support user-defined scripts such as npm build, or recommend how to run them via Composer
+
+    $this->logger->info("Mirroring source files from {$this->dir} to $artifact_dir");
+    $originFinder = Finder::create();
+    $originFinder->files()->in($this->dir)
+      // Include dot files like .htaccess.
+      ->ignoreDotFiles(FALSE)
+      // Ignore VCS ignored files (e.g. vendor) to speed up the mirror (Composer will restore them later).
+      ->ignoreVCSIgnored(TRUE)
+      // Ignore .gitignore files (everything should be committed to the artifact).
+      ->notName('.gitignore');
+    $targetFinder = Finder::create();
+    $targetFinder->files()->in($artifact_dir)->ignoreDotFiles(FALSE);
+    $this->localMachineHelper->getFilesystem()->mirror($this->dir, $artifact_dir, $originFinder, ['override' => TRUE, 'delete' => TRUE], $targetFinder);
+
+    $this->logger->info('Installing Composer production dependencies');
+    $this->localMachineHelper->executeFromCmd('composer install --no-dev --no-interaction --optimize-autoloader', $output_callback, $artifact_dir, $this->output->isVerbose());
+  }
+
+  protected function sanitize(string $artifact_dir):void {
```

Comment:
We may also want to make sanitization optional by adding a `--no-sanitize` flag.